samples/csharp/getting-started/DeepLearning_ObjectDetection_Onnx/ObjectDetectionConsoleApp/OnnxModelScorer.cs

```diff
@@ -0,0 +1,100 @@
+            this.imagesFolder = imagesFolder;
+            this.modelLocation = modelLocation;
+            mlContext = new MLContext();
+        }
+
+        public struct ImageNetSettings
+        {
+            public const int imageHeight = 416;
+            public const int imageWidth = 416;
+            public const bool channelsLast = true;
+        }
+
+        public struct TinyYoloModelSettings
+        {
+            // for checking TIny yolo2 Model input and  output  parameter names,
+            //you can use tools like Netron, 
+            // which is installed by Visual Studio AI Tools
+
+            // input tensor name
+            public const string ModelInput = "image";
+
+            // output tensor name
+            public const string ModelOutput = "grid";
+        }
+
+        public void Score()
+        {
+            var model = LoadModel(imagesLocation, imagesFolder, modelLocation);
+
+            PredictDataUsingModel(imagesLocation, imagesFolder, model);
+        }
+
+        private PredictionEngine<ImageNetData, ImageNetPrediction> LoadModel(string imagesLocation, string imagesFolder, string modelLocation)
+        {
+            Console.WriteLine("Read model");
+            Console.WriteLine($"Model location: {modelLocation}");
+            Console.WriteLine($"Images folder: {imagesFolder}");
+            Console.WriteLine($"Training file: {imagesLocation}");
+            Console.WriteLine($"Default parameters: image size=({ImageNetSettings.imageWidth},{ImageNetSettings.imageHeight})");
+
+            var data = mlContext.Data.LoadFromTextFile<ImageNetData>(imagesLocation, hasHeader: true);
+
+            var pipeline = mlContext.Transforms.LoadImages(outputColumnName: "image", imageFolder: imagesFolder, inputColumnName: nameof(ImageNetData.ImagePath))
+                            .Append(mlContext.Transforms.ResizeImages(outputColumnName: "image", imageWidth: ImageNetSettings.imageWidth, imageHeight: ImageNetSettings.imageHeight, inputColumnName: "image"))
+                            .Append(mlContext.Transforms.ExtractPixels(outputColumnName: "image"))
+                            .Append(mlContext.Transforms.ApplyOnnxModel(modelFile: modelLocation, outputColumnNames: new[] { TinyYoloModelSettings.ModelOutput }, inputColumnNames: new[] { TinyYoloModelSettings.ModelInput }));
+
+            var model = pipeline.Fit(data);
+
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<ImageNetData, ImageNetPrediction>(model);
+
+            return predictionEngine;
+        }
+
+        protected void PredictDataUsingModel(string imagesLocation,
+                                                                  string imagesFolder,
+                                                                  PredictionEngine<ImageNetData, ImageNetPrediction> model)
+        {
+            Console.WriteLine("Identify the objects in the images");
+            Console.WriteLine("");
```

Comment:
Instead of printing comments here, is it easy to return the predictions values to the main() function and add all comments there? It makes the example more readable.

Reply:
That's good approach. But currently I have followed the existing samples. For Ex: TensorFlow_Imagescorer is printing the output in the prediction method. I think we need to change it for all the samples. 

@CESARDELATORRE Do you also suggest to change this?